design/FY2017/NFP-Single-Zone VAV Fan Control for PackagedTerminalUnits.md

```diff
@@ -0,0 +1,510 @@
+
+Lines 7126-7531 of HVACUnitarySystem.cc contain the latest model for SZVAV. This code will be moved to a unique function and revised to make all equipment specific data available for read/write.
+
+For example:
+
+The code at lines 7126-7531 will be moved to function CalcSZVAVModel. The function parameters for input and output will be passed using arrays. A reference to the equipment model array will also be passed to allow read/write of data to specific equipment models. The code in UnitarySystem starts like this. As you can see, the member variables are already set to local variables for use in the model. This is good and can preface the call to the common function in any equipment model. These variables will be written to the SZVAVInputs array as discussed later.
+
+		// use the ASHRAE 90.1 method of reduced fan speed at low loads
+		if ( UnitarySystem( UnitarySysNum ).simASHRAEModel ) {
+
+			// set up mode specific variables to use in common function calls
+			if ( CoolingLoad ) {
+				maxCoilFluidFlow = UnitarySystem( UnitarySysNum ).MaxCoolCoilFluidFlow;
+				maxOutletTemp = UnitarySystem( UnitarySysNum ).DOASDXCoolingCoilMinTout;
+				minAirMassFlow = UnitarySystem( UnitarySysNum ).MaxNoCoolHeatAirMassFlow;
+				maxAirMassFlow = UnitarySystem( UnitarySysNum ).MaxCoolAirMassFlow;
+				lowSpeedFanRatio = UnitarySystem( UnitarySysNum ).LowSpeedCoolFanRatio;
+				coilFluidInletNode = UnitarySystem( UnitarySysNum ).CoolCoilFluidInletNode;
+				coilFluidOutletNode = UnitarySystem( UnitarySysNum ).CoolCoilFluidOutletNodeNum;
+				coilLoopNum = UnitarySystem( UnitarySysNum ).CoolCoilLoopNum;
+
+However, there are places where information needs to be passed to the model in other locations within the equipment model code. For example, in this section of the existing SZVAV model, a member variable (*CoilWaterFlowRatio) is set so it can be used elsewhere:
+
+			if ( CoolingLoad ) { // Function CalcUnitarySystemToLoad, 4th and 5th arguments are CoolPLR and HeatPLR
+				// set the water flow ratio so water coil gets proper flow
+				UnitarySystem( UnitarySysNum ).CoolCoilWaterFlowRatio = maxCoilFluidFlow / UnitarySystem( UnitarySysNum ).MaxCoolCoilFluidFlow;
+				CalcUnitarySystemToLoad( UnitarySysNum, AirLoopNum, FirstHVACIteration, PartLoadRatio, 0.0, OnOffAirFlowRatio, TempSensOutput, TempLatOutput, HXUnitOn, _, _, CompressorONFlag );
+			} else {
+				UnitarySystem( UnitarySysNum ).HeatCoilWaterFlowRatio = maxCoilFluidFlow / UnitarySystem( UnitarySysNum ).MaxHeatCoilFluidFlow;
+				CalcUnitarySystemToLoad( UnitarySysNum, AirLoopNum, FirstHVACIteration, 0.0, PartLoadRatio, OnOffAirFlowRatio, TempSensOutput, TempLatOutput, HXUnitOn, _, _, CompressorONFlag );
+			}
+
+
+To write to equipment model member variables, a reference to the member array would need to be passed to the function.
+
+    auto & SZVAVModel( FanCoil[ FanCoilNum ] );
+    auto & SZVAVModel( UnitarySystem[ UnitarySystemNum ] );
+    auto & SZVAVModel( PTUnit[ PTUnitNum ] );
```

Comment:
Hmmm, I think we'll need to chat about this approach.  I'm not sure this will work in the way you are wanting it to work, but 1) I may be missing something and 2) I think we can definitely find a way that _will_ work.

Reply:
I agree.